hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -506,13 +506,15 @@ public OutputStream createFile(final Path path,
         triggerConditionalCreateOverwrite = true;
       }
 
+      AbfsLease lease = maybeCreateFiniteLease(relativePath, isNamespaceEnabled);
```

Comment:
If the path qualifies for infinteLease as per config settings, this will still create a finite lease ? 

Call to a method createLease() that will first check for infiniteLease setting before defaulting to finite lease (if write lease config is enabled), would be the expectation ?

Reply:
No it will not, it returns a null object if path qualifies for infinite lease